com.microsoft.mrtk.spatialmanipulation/ObjectManipulator/NewObjectManipulator.cs

```diff
@@ -0,0 +1,575 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Unity.Profiling;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.Serialization;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace Microsoft.MixedReality.Toolkit.SpatialManipulation
+{
+    /// <summary>
+    /// ObjectManipulator allows for the manipulation (move, rotate, scale)
+    /// of an object by any interactor with a valid attach transform.
+    /// Multi-handed interactions and physics-enabled objects are also supported.
+    /// </summary>
+    /// <remarks>
+    /// ObjectManipulator works with both rigidbody and non-rigidbody objects,
+    /// and allows for throwing and catching interactions. Any interactor
+    /// with a well-formed attach transform can interact with and manipulate
+    /// an ObjectManipulator. This is a drop-in replacement for the built-in
+    /// XRI XRGrabInteractable, that allows for flexible multi-handed interactions.
+    /// ObjectManipulator doesn't track controller velocity, so for precise fast-paced
+    /// throwing interactions that only need one hand, XRGrabInteractable may
+    /// give better results.
+    /// </remarks>
+    [RequireComponent(typeof(PlacementHub))]
+    [AddComponentMenu("MRTK/Spatial Manipulation/New Object Manipulator")]
+    public class NewObjectManipulator : StatefulInteractable
```

Comment:
Mark internal, probably. I don't think we should stay with this naming. If we need to have the old and new objmanips side by side (I hope not) we should name it something new, like `PlacementManipulator`. Just want to avoid having public API surface that's explicitly named "NewThing"

Reply:
I think after this naming hump, I want the "NewObjectManipulator" to be public, much like our previous one was. We've had prior examples of people extending our base manipulator, and I don't want to buck that expectation, since our other UI components are also marked as public currently.